src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -337,267 +214,210 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
         /// <inheritdoc />
         public override async Task<PackageMetadata> GetPackageMetadata(PackageURL purl, bool useCache = true)
         {
-            PackageMetadata metadata = new();
-            string? content = await GetMetadata(purl, useCache);
-            if (string.IsNullOrEmpty(content)) { return metadata; }
+            string latestVersion = (await EnumerateVersions(purl, useCache)).First();
 
-            // convert NuGet package data to normalized form
-            JsonDocument contentJSON = JsonDocument.Parse(content);
-            JsonElement root = contentJSON.RootElement;
-            JsonElement latestCatalogEntry = GetLatestCatalogEntry(root);
+            // Construct a new PackageURL that's guaranteed to have a version, the latest version is used if no version was provided.
+            PackageURL purlWithVersion = !string.IsNullOrWhiteSpace(purl.Version) ? 
+                purl : new PackageURL(purl.Type, purl.Namespace, purl.Name, latestVersion, purl.Qualifiers, purl.Subpath);
 
-            metadata.Name = latestCatalogEntry.GetProperty("id").GetString();
-            metadata.Description = latestCatalogEntry.GetProperty("description").GetString();
-            
-            // Title is different than name or description for NuGet packages, not always used.
-            // metadata.Title = GetLatestCatalogEntry(root).GetProperty("title").GetString();
+            NuGetMetadata? packageVersion =
+                await _nuGetProvider.GetMetadataAsync(purlWithVersion, useCache: useCache);
+
+            if (packageVersion is null)
+            {
+                throw new NullReferenceException();
+            }
```

Comment:
Nit: you can compress null checks like this using the null coalescing operator `??`:

```suggestion
            NuGetMetadata packageVersion =
                await _nuGetProvider.GetMetadataAsync(purlWithVersion, useCache: useCache) ?? throw new InvalidOperationException("Could not get metadata for {PURL}", purlWithVersion);
```

Reply:
Resolved in other PR.